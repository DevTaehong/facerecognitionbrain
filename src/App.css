.App {
  text-align: center;
}

.center {
  display: flex;
  justify-content: center;
}

.particles-bg-canvas-self {
  position: absolute;
  width: 100%;
  height: 150%;
  left: 0;
  top: 0;
  z-index: -1;
  
}