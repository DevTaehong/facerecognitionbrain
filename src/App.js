import React, { Component } from 'react';
import './App.css';
import Navigation from './components/Navigation/Navigation';
import Logo from './components/Logo/Logo';
import ImageLinkForm from './components/ImageLinkForm/ImageLinkForm';
import FaceRecognition from './components/FaceRecognition/FaceRecognition';
import Rank from './components/Rank/Rank';
// import Particle from './components/Particle/Particle'
import ParticlesBg from 'particles-bg';
import Clarifai from 'clarifai';
import Signin from './components/Signin/Signin';
import Register from './components/Register/Register';

const app = new Clarifai.App({
  apiKey: 'b58172beddae4925b565b284608ad97d'
});
class App extends Component {
  constructor(){
    super();
    this.state = {
      input: '',
      imageUrl: '',
      box: {},
      route: 'signin',
      isSignedIn: false
    }
  }

  calculateFaceLocation = (data) => {
    const clarifaiFace = data.outputs[0].data.regions[0].region_info.bounding_box;
    const image = document.getElementById('inputImage');
    const width = Number(image.width);
    const height = Number(image.height);
    return { // We will setState box{}
      leftCol: clarifaiFace.left_col * width,
      topRow: clarifaiFace.top_row * height,
      rightCol: width - (clarifaiFace.right_col * width),
      bottomRow: height - (clarifaiFace.bottom_row * height)
    }
  }
  
  disPlayFaceBox = (box) => {
    this.setState({box: box}); // You can just use this.setState({box}) because of ES6
  }

  onInputChange = (event) => {
    this.setState({input: event.target.value});
  }
  
  onButtonSubmit = () => {
    this.setState({imageUrl: this.state.input})
    app.models.predict(Clarifai.FACE_DETECT_MODEL, this.state.input) // this.state.imageUrl cannot work here because of how this.setState works.
    .then((response) => this.disPlayFaceBox(this.calculateFaceLocation(response)))
    .catch(err => console.log(err));
  }

  onRouteChange = (route) => {
    if (route === 'signout') {
      this.setState({isSignedIn: false});
    } else if (route === 'home') {
      this.setState({isSignedIn: true});
    }
    this.setState({route: route}); //Should warp this in curly brackets because it's an object
  }

  render() {
    const { isSignedIn, imageUrl, route, box } = this.state;
    return (
      <div className="App">
        <ParticlesBg type="circle" bg={true} />
        <Navigation isSignedIn={isSignedIn} onRouteChange={this.onRouteChange}/>
        { route === 'home' 
          ? <div> 
              <Logo />
              <Rank />
              <ImageLinkForm 
                onInputChange={this.onInputChange} 
                onButtonSubmit={this.onButtonSubmit}
              />
              <FaceRecognition box={box} imageUrl={imageUrl}/>
            </div>
          : (
              route === 'signin'
              ? <Signin onRouteChange={this.onRouteChange} />
              : <Register onRouteChange={this.onRouteChange} /> 
            )
        }
      </div>
    );
  }
}

export default App;
