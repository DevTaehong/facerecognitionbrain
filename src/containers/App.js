import React, { Component } from 'react';
import './App.css';
import Navigation from '../components/Navigation/Navigation';
import Logo from '../components/Logo/Logo';
import ImageLinkForm from '../components/ImageLinkForm/ImageLinkForm';
import FaceRecognition from '../components/FaceRecognition/FaceRecognition';
import Rank from '../components/Rank/Rank';
import ParticlesBg from 'particles-bg';
import Signin from '../components/Signin/Signin';
import Register from '../components/Register/Register';

const initialState = {
  input: '',
  imageUrl: '',
  box: {},
  route: 'signin',
  isSignedIn: false,
  imageHeight: 0,
  user: {
    id: '', 
    name: '', 
    email: '', 
    entries: 0, 
    joined: ''
  }
}
class App extends Component {
  constructor(){
    super();
    this.state = initialState;
  }


  loadUser = (data) => {
    this.setState({
      user:{
        id: data.id, 
        name: data.name, 
        email: data.email, 
        entries: data.entries, 
        joined: data.joined
      }
    });
  }

  calculateFaceLocation = (data) => {
    const clarifaiFace = data.outputs[0].data.regions[0].region_info.bounding_box;
    const image = document.getElementById('inputImage');
    const width = Number(image.width);
    const height = Number(image.height);
    this.setState({imageHeight: height})
    return { // We will do setState box{}
      leftCol: clarifaiFace.left_col * width,
      topRow: clarifaiFace.top_row * height,
      rightCol: width - (clarifaiFace.right_col * width),
      bottomRow: height - (clarifaiFace.bottom_row * height)
    }
  }
  
  disPlayFaceBox = (box) => {
    this.setState({box: box}); // You can just use this.setState({box}) because of ES6
  }

  onInputChange = (event) => {
    this.setState({input: event.target.value});
  }
  
  onPictureSubmit = () => {
    this.setState({imageUrl: this.state.input})
    fetch('https://whispering-spire-95505.herokuapp.com/imageurl', { 
            method: 'post',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ // It doesn't understand JavaScript, so change it to JSON 
                input: this.state.input
            })
    })
    .then(response => response.json())
    .then(response => {
      if (response) {
        fetch('https://whispering-spire-95505.herokuapp.com/image', { 
            method: 'put',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ // It doesn't understand JavaScript, so change it to JSON 
                id: this.state.user.id
            })
        })
        .then(response => response.json())
        .then(count => {
          this.setState(Object.assign(this.state.user, { entries: count }))
        })
        .catch(console.log)
      }
      this.disPlayFaceBox(this.calculateFaceLocation(response))
    })
    .catch(err => console.log(err));
  }

  onRouteChange = (route) => {
    if (route === 'signout') {
      this.setState(initialState);
    } else if (route === 'home') {
      this.setState({isSignedIn: true});
    }
    this.setState({route: route}); //Should warp this in curly brackets because it's an object
  }

  render() {
    const { isSignedIn, imageUrl, route, box, user } = this.state;
    return (
      <div className="App">
        { this.state.imageHeight === 0 
          ? <ParticlesBg 
            type="circle"
            bg={{
              position: "absolute",
              width: 100+"%",
              height: 100+"%",
              left: 0,
              top: 0,
              zIndex: -1
            }}
          /> 
          : 
          <ParticlesBg 
            type="circle"
            bg={{
              position: "absolute",
              width: 100+"%",
              height: 100+(this.state.imageHeight/10)+"%",
              left: 0,
              top: 0,
              zIndex: -1
            }}
          />
        }
        <Navigation isSignedIn={isSignedIn} onRouteChange={this.onRouteChange} />
        { route === 'home' 
          ? <div> 
              <Logo />
              <Rank name={user.name} entries={user.entries} />
              <ImageLinkForm 
                onInputChange={this.onInputChange} 
                onPictureSubmit={this.onPictureSubmit}
              />
              <FaceRecognition box={box} imageUrl={imageUrl} />
            </div>
          : (
              route === 'signin' 
              ? <Signin loadUser={this.loadUser} onRouteChange={this.onRouteChange} />
              : ( route === 'signout' 
                ? <Signin loadUser={this.loadUser} onRouteChange={this.onRouteChange} /> 
                : <Register loadUser={this.loadUser} onRouteChange={this.onRouteChange} />)
            )
        }
      </div>
    );
  }
}

export default App;
