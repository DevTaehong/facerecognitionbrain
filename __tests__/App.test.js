/**
 * @jest-environment jsdom
 */
import App from '../src/containers/App';
import renderer from 'react-test-renderer';


it("renders correctly", () => {
    const tree = renderer
    .create(<App />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

