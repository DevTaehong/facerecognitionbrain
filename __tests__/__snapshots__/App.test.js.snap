// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <canvas
    className="particles-bg-canvas-self"
    onMouseDown={[Function]}
    style={
      Object {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": -1,
      }
    }
  />
  <nav
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-end",
      }
    }
  >
    <p
      className="f3 link dim white pa3 pv2 mv1 mr3 pointer"
      onClick={[Function]}
    >
      Log In
    </p>
    <p
      className="f3 link dim white pa3 pv2 mv1 mr3 pointer"
      onClick={[Function]}
    >
      Sign up
    </p>
  </nav>
  <div
    className="login-form mv4 w-100 w-50-m w-25-l mw6"
  >
    <div>
      <h1>
        Login
      </h1>
      <div
        className="content"
      >
        <div
          className="input-field"
        >
          <input
            id="email-address"
            name="email-address"
            onChange={[Function]}
            placeholder="Email"
            type="email"
          />
        </div>
        <div
          className="input-field"
        >
          <input
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
      <div
        className="action"
      >
        <button
          onClick={[Function]}
        >
          Log in
        </button>
        <button
          onClick={[Function]}
        >
          Sign up
        </button>
      </div>
    </div>
  </div>
</div>
`;
